src/scripts/syncBreaches.js

```diff
@@ -11,7 +11,83 @@
 
 import { req, formatDataClassesArray } from '../utils/hibp.js'
 import { getAllBreaches, upsertBreaches } from '../db/tables/breaches.js'
+import { mkdir, readdir } from "node:fs/promises";
+import { resolve as pathResolve } from "node:path";
+import { finished } from 'stream/promises';
+import fs from "fs";
+import { Readable } from 'stream';
+import aws from 'aws-sdk';
+import { Upload } from "@aws-sdk/lib-storage";
 
+// Get breaches logos and uploads to s3
+const accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+const region = process.env.AWS_REGION;
+const Bucket = process.env.S3_BUCKET;
+
+const s3 = new aws.S3({
+  region,
+  credentials: {
+    accessKeyId,
+    secretAccessKey,
+  },
+});
+
+async function uploadToS3(fileName, fileStream) {
+  const uploadParams = {
+    Bucket,
+    Key: fileName,
+    Body: fileStream
+  }
+  try {
+  await s3.upload(uploadParams).promise()
+  console.log('Successfully uploaded data to ' + Bucket + '/' + fileName)
+} catch (err) {
+  console.error(err, err.stack)
+}
+}
+
+export async function getBreachIcons(breaches) {
+  let logoMap;
+  async function fetchBreachIcons() {
```

Comment:
nit: not sure if we have a style guide or not, but sometimes nested function definitions can get confusing.
Can we move this up to the root level (since I think it's hoisted in scope anyways) or are we depending on some leaky variables or something?

Reply:
Taken out. thanks!